iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl

```diff
@@ -502,6 +502,40 @@ string<Capacity>::append(TruncateToCapacity_t, const T& t) noexcept
     return *this;
 }
 
+template <uint64_t Capacity>
+template <typename T>
+inline constexpr typename std::enable_if<is_char_array<T>::value || is_cxx_string<T>::value, bool>::type
+string<Capacity>::insert(const uint64_t pos, const T& str, const uint64_t count) noexcept
+{
+    if (count > internal::GetSize<T>::call(str))
+    {
+        return false;
+    }
+    const auto new_size = m_rawstringSize + count;
+    if (new_size > Capacity)
+    {
+        return false;
+    }
+    if (Capacity < pos + count)
+    {
+        return false;
+    }
+
+    auto remainder = substr(pos);
+    if (!remainder)
+    {
+        return false;
+    }
+
+    std::memcpy(&(m_rawstring[pos]), internal::GetData<T>::call(str), count);
+    std::memcpy(&(m_rawstring[pos + count]), remainder->c_str(), remainder->size());
+
+    m_rawstring[new_size] = '\0';
```

Comment:
Not sure how the static checks (that fail in the pipeline with a specific gcc) work but we could move setting the zero-terminator before the memcpy operations. The static checks should not really know about where the zero-terminator will be (maybe that it directly follows the remaineder copy, but definitely not the exact position as it is runtime info).